File: app/src/main/java/com/magic/upcoming/games/network/exception/ResponseFailedException.java
package com.magic.upcoming.games.network.exception;

/**
 * Created by ThinkPad on 2017/11/2.
 */

public class ResponseFailedException extends Throwable {

    private RxError rxError;
    public ResponseFailedException(){}
    public ResponseFailedException(int errorCode, String mes)
    {
        rxError=new RxError(errorCode,mes);
    }
    public ResponseFailedException(int errorCode, String mes, Object data)
    {
        rxError=new RxError(errorCode,mes,data);
    }
    public RxError getRxError() {
        return this.rxError;
    }
}

File: app/src/main/java/com/magic/upcoming/games/network/exception/ResponseFailedException.kt
package com.magic.upcoming.games.network.exception

/**
 * Created by ThinkPad on 2017/11/2.
 */
class ResponseFailedException : Throwable {

    var errorMessage: String? = null

    constructor()

    constructor(mes: String?) {
        errorMessage = mes
    }

    override val message: String?
        get() = if (errorMessage.isNullOrEmpty()) super.message
                    else errorMessage

}